UndertaleModTool/Editors/UndertaleShaderEditor.xaml

```diff
@@ -0,0 +1,104 @@
+            </ObjectDataProvider.MethodParameters>
+        </ObjectDataProvider>
+    </UserControl.Resources>
+    <Grid>
+        <Grid.ColumnDefinitions>
+            <ColumnDefinition Width="1*"/>
+            <ColumnDefinition Width="3*"/>
+        </Grid.ColumnDefinitions>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+
+        <TextBlock Grid.Row="0" Grid.Column="0" Margin="3">Name</TextBlock>
+        <local:UndertaleStringReference Grid.Row="0" Grid.Column="1" Margin="3" ObjectReference="{Binding Name}"/>
+
+        <TextBlock Grid.Row="1" Grid.Column="0" Margin="3">Type</TextBlock>
+        <ComboBox ItemsSource="{Binding Source={StaticResource ShaderType}}" SelectedItem="{Binding Path=Type}"
+                  Grid.Row="1" Grid.Column="1" Margin="3"/>
+
+        <TextBlock Grid.Row="2" Grid.Column="0" Margin="3">GLSL ES vertex source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="2" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding GLSL_ES_Vertex}"/>
+
+        <TextBlock Grid.Row="3" Grid.Column="0" Margin="3">GLSL ES fragment source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="3" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding GLSL_ES_Fragment}"/>
+
+        <TextBlock Grid.Row="4" Grid.Column="0" Margin="3">GLSL vertex source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="4" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding GLSL_Vertex}"/>
+
+        <TextBlock Grid.Row="5" Grid.Column="0" Margin="3">GLSL fragment source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="5" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding GLSL_Fragment}"/>
+
+        <TextBlock Grid.Row="6" Grid.Column="0" Margin="3">HLSL9 vertex source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="6" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding HLSL9_Vertex}"/>
+
+        <TextBlock Grid.Row="7" Grid.Column="0" Margin="3">HLSL9 fragment source</TextBlock>
+        <local:UndertaleStringReference Grid.Row="7" Grid.Column="1" Margin="3" Height="40" ObjectReference="{Binding HLSL9_Fragment}"/>
+
+        <Border Grid.Row="8" Grid.Column="0" Margin="20" Background="LightPink" BorderBrush="Red" BorderThickness="3">
+            <TextBlock TextWrapping="Wrap" Margin="3" Foreground="Red">NOTE: The vertex shader attributes editor is currently buggy and may not work as expected.</TextBlock>
+        </Border>
+        
+        <!-- TODO: Fix this editor -->
```

Comment:
I guess I'll know what 'fix' means when I try to run this but this comment could probably be a little bit more specific

Reply:
Now I see that you wrote what it means in the pull request itself ;)